nanofi/include/core/cstructs.h

```diff
@@ -136,6 +141,12 @@ typedef struct file_buffer {
   uint64_t file_len;
 } file_buffer;
 
-typedef struct cstream cstream;
+#ifndef _WIN32
```

Comment:
https://stackoverflow.com/questions/662084/whats-the-difference-between-the-win32-and-win32-defines-in-c

It's not really recommended to use WIN32 without the underscore.

Reply:
Hmm _WIN32 can be compiler specific and potentially not defined appropriately on all compilers (but I don't know if we will ever hit it with mingw or any others ), to be safe we define WIN32 via CMAKE which does the compiler general checks. While this is against the norm in the project, I don't have any strong feelings in nanofi. It will may be broken apart and have different conventions.